app/src/main/java/com/android/userinfo/UserInputActivity.kt

```diff
@@ -0,0 +1,146 @@
+        val editTextPhone = binding.editTextViewPhone
+        val editTextPincode = binding.editTextViewPincode
+        val editTextAddress = binding.editTextViewAddress
+
+        val cancelConfirmButtonLayout = binding.buttonLayout
+        val cancelButton = binding.buttonCancel
+        val confirmButton = binding.buttonConfirm
+        val buttonValidate = binding.buttonValidate
+        val layout = binding.constraintLayout
+
+        buttonValidate.setOnClickListener(object : View.OnClickListener {
+            override fun onClick(view: View) {
+                val isAllFieldsFilled = validateAllFields(layout)
+                if (isAllFieldsFilled
+                    && validatePhoneField(editTextPhone)
+                    && validatePincodeField(editTextPincode,this@UserInputActivity)
+                    && validateEmailField(editTextEmail)
+                ) {
+                    buttonValidate.visibility = View.INVISIBLE
+                    cancelConfirmButtonLayout.visibility = View.VISIBLE
+                    disableEditViews()
+                } else {
+                    startActivity(getIntent())
+                }
+            }
+
+            private fun disableEditViews() {
+                for (i in 0 until layout.childCount) {
+                    val v = layout.getChildAt(i)
+                    if (v is EditText) {
+                        v.isEnabled = false
+                        v.setTextColor(Color.BLACK)
+                    }
+                }
+            }
+        })
+
+        confirmButton.setOnClickListener(object : View.OnClickListener {
+            override fun onClick(view: View) {
+                val bundle = Bundle()
+                bundle.putString(USERNAME, editTextUsername.text.toString())
+                bundle.putString(EMAIL, editTextEmail.text.toString())
+                bundle.putString(PHONE, editTextPhone.text.toString())
+                bundle.putString(PINCODE, editTextPincode.text.toString())
+                bundle.putString(ADDRESS, editTextAddress.text.toString())
+
+                val intent = Intent(this@UserInputActivity, InfoDisplayActivity::class.java).apply {
+                    putExtras(bundle)
+                }
+                startActivity(intent)
+            }
+        })
+
+        cancelButton.setOnClickListener(object : View.OnClickListener {
+            override fun onClick(view: View) {
+                buttonValidate.visibility = View.VISIBLE
+                cancelConfirmButtonLayout.visibility = View.INVISIBLE
+                enableEditViews()
+            }
+
+            private fun enableEditViews() {
+                for (i in 0 until layout.childCount) {
+                    val v = layout.getChildAt(i)
+                    if (v is EditText) {
+                        v.isEnabled = true
+                    }
+                }
+            }
+        })
+    }
+
+    private fun validateEmailField(view: EditText): Boolean {
```

Comment:
We can move validating to another class. makes us easier to write local tests
we should not pass views normally. we can pass email and validate it